File: 04.Simulation模拟/leetcode_922_sortArrayByParityII.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 0);
        int odd = 1, even = 0;
        for (int& i : nums){
            if (i&1){
                res[odd] = i;
                odd += 2;
            }
            else{
                res[even] = i;
                even += 2;
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {4,2,5,7};
    vector<int> res = s.sortArrayByParityII(nums);
    for (auto& i : res){
        cout << i << " ";
    }
    cout << endl;
}

File: 04.Simulation模拟/leetcode_922_sortArrayByParityII.py
from typing import List

class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0]*n
        odd, even = 1, 0
        for i in nums:
            if i&1:
                res[odd] = i
                odd += 2
            else:
                res[even] = i
                even += 2
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [4,2,5,7]
    print(s.sortArrayByParityII(nums))
